paddleseg/models/enet.py

```diff
@@ -0,0 +1,759 @@
+        # #         output_size = output_size[2:]
+        # t = str(input.dtype).lower().strip().split(".")[-1]
+        # # print(t)
+        # t = TYPE_MAPPER[t]
+        # out = paddle.zeros(output_size, dtype=t)
+        # flatten_out = paddle.flatten(out)
+        # pdb.set_trace()
+        # for i in range(indices.shape[0]):
+        #     for j in range(indices.shape[1]):
+        #         for k in range(indices.shape[2]):
+        #             for m in range(indices.shape[3]):
+        #                 indices[i, j, k, m] = (out.shape[1] * out.shape[2] * out.shape[3]) * i + \
+        #                                       (out.shape[2] * out.shape[3]) * j + indices[i, j, k, m]
+        # flatten_indices = paddle.flatten(indices)
+        # flatten_input = paddle.flatten(input)
+        # for i in range(flatten_indices.shape[0]):
+        #     flatten_out[flatten_indices[i].tolist()] = flatten_input[i].tolist()
+        # out = paddle.reshape(flatten_out, out.shape)
+        return out
+
+
+class InitialBlock(nn.Layer):
+    """The initial block is composed of two branches:
+    1. a main branch which performs a regular convolution with stride 2;
+    2. an extension branch which performs max-pooling.
+    Doing both operations in parallel and concatenating their results
+    allows for efficient downsampling and expansion. The main branch
+    outputs 13 feature maps while the extension branch outputs 3, for a
+    total of 16 feature maps after concatenation.
+    Keyword arguments:
+    - in_channels (int): the number of input channels.
+    - out_channels (int): the number output channels.
+    - kernel_size (int, optional): the kernel size of the filters used in
+    the convolution layer. Default: 3.
+    - padding (int, optional): zero-padding added to both sides of the
+    input. Default: 0.
+    - bias (bool, optional): Adds a learnable bias to the output if
+    ``True``. Default: False.
+    - relu (bool, optional): When ``True`` ReLU is used as the activation
+    function; otherwise, PReLU is used. Default: True.
+    """
+    def __init__(self, in_channels, out_channels, bias=False, relu=True):
+        super(InitialBlock, self).__init__()
+
+        if relu:
+            activation = nn.ReLU
+        else:
+            activation = nn.PReLU
+
+        # Main branch - As stated above the number of output channels for this
+        # branch is the total minus 3, since the remaining channels come from
+        # the extension branch
+        self.main_branch = nn.Conv2D(in_channels,
+                                     out_channels - 3,
+                                     kernel_size=3,
+                                     stride=2,
+                                     padding=1,
+                                     bias_attr=bias)
+
+        # Extension branch
+        self.ext_branch = nn.MaxPool2D(3, stride=2, padding=1)
+
+        # Initialize batch normalization to be used after concatenation
```

Comment:
Annotations like this is not necessary, plz remove it

Reply:
Done.